src/network/serverpackethandler.cpp

```diff
@@ -471,6 +471,12 @@ void Server::process_PlayerPos(RemotePlayer *player, PlayerSAO *playersao,
 	v3f position((f32)ps.X / 100.0f, (f32)ps.Y / 100.0f, (f32)ps.Z / 100.0f);
 	v3f speed((f32)ss.X / 100.0f, (f32)ss.Y / 100.0f, (f32)ss.Z / 100.0f);
 
+	const v3s16 old_node_position = floatToInt(playersao->getBasePosition(), BS);
+	const v3s16 new_node_position = floatToInt(position, BS);
+
+	if(old_node_position != new_node_position)
+		m_script->on_moveplayer(playersao, new_node_position, old_node_position);
```

Comment:
please compare versions before converting and convert when sending this call, this pervent useless converts

Reply:
Yes, I'm comparing them after converting to reduce the number of calls (if I got what you are saying correctly)

Reply:
the number of calls to on_moveplayer, that is.

Reply:
Yes the conversions need to be done before comparing, so it's fine.